core/src/main/java/com/datastrato/gravitino/storage/relational/session/SqlSessions.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.storage.relational.session;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.TransactionIsolationLevel;
+
+/**
+ * SqlSessions is a utility class to maintain the MyBatis's {@link SqlSession} object. It is a
+ * thread local class and should be used to get the {@link SqlSession} object. It also provides the
+ * methods to commit, rollback and close the {@link SqlSession} object.
+ */
+public final class SqlSessions {
+  private static final ThreadLocal<SqlSession> sessions = new ThreadLocal<>();
+
+  private SqlSessions() {}
+
+  @VisibleForTesting
+  static ThreadLocal<SqlSession> getSessions() {
```

Comment:
Why do we need to use ThreadLocal here?

Reply:
Because if different threads use the same `SqlSession`, they will affect each other and the independence of transactions cannot be guaranteed.

Reply:
Could we new a session for every request?

Reply:
> Could we new a session for every request?

Do you mean that when requesting the database, each request uses a new `SqlSession`? If so, transaction management will become more difficult. For example, when deleting, we may need to initiate multiple requests for cascading deletion. These operations are not in the same `SqlSession`, then we need to obtain all involved `SqlSession` for rollback operation.
In addition, each time we create a new `SqlSession`, it need to re-establish a connection with the database, which will cause a loss in performance under high QPS conditions.
Therefore, the `ThreadLocal` method is adopted to ensure that the same `SqlSession` is reused in a single thread, reducing the loss of creating a `SqlSession`, and making it easier for us to manage transactions.

Reply:
The request which I mentioned means that a Gravitino request instead of database request. 
For ThreadLocal, could you add some tests to verify the clean of thread local variable.

Reply:
> The request which I mentioned means that a Gravitino request instead of database request. For ThreadLocal, could you add some tests to verify the clean of thread local variable.

Already add this part of unit tests in `TestSqlSession` class

Reply:
@jerryshao For now, there will be no problem. We use the `try-with-resource` to ensure that the database connection of `SqlSession` is closed after its use is completed, and the `finally` statement is used when clearing the thread local variable. Even if it is not cleaned and reused successfully, the same `SqlSession` will be used to open a new database connection.